modules/actor/actor-wfrp4e.js

```diff
@@ -3380,14 +3389,14 @@ export default class ActorWfrp4e extends Actor {
 
   }
 
-  deleteEffectsFromItem(itemId) {
+  async deleteEffectsFromItem(itemId) {
     let removeEffects = this.effects.filter(e => {
       if (!e.origin)
         return false
       return e.origin.includes(itemId)
     }).map(e => e.id).filter(id => this.actorEffects.has(id))
 
-    this.deleteEmbeddedDocuments("ActiveEffect", removeEffects)
+    await this.deleteEmbeddedDocuments("ActiveEffect", removeEffects)
```

Comment:
probably not, again, its my preference comming from .NET world, to await the promise/task instead returning it, especially when there are no visible signatures of the functions (i.e. like in Typescript). If at least there would be async postfix in each method that returns promise - it would be more readable.